target eda

danpereda opened the pull request:
Avance con EDA al target,

Se nota que el corte en las fechas no es aleatorio, es posible abusar de esto de alguna manera?

Por otro lado la mayoría de las variables a predecir son estacionarias.

.py es lo mismo pero leíble en github.

danpereda committed:
target eda

datacubeR commented:
![image](https://user-images.githubusercontent.com/48638337/152899935-748cc8f4-7088-49b6-b222-e595990febfc.png)

Definitivamente esta distribución de registros no se ve al azar. Ahora, creo que para el test se predicen test_ids en el futuro, por lo que esperaría que la distribución de los últimos test_id es la que debiera prevalecer. 

Como conclusión  de acá yo saco en limpio que quizás no vale la pena entrenar con toda la data, sino con alguna porción del final?

Respecto a los test y el UMAP no sé si pudiste sacar alguna conclusión...

datacubeR commented on notebooks_py/EDA_Target.py:
```diff
@@ -0,0 +1,160 @@
+pd.Series(orders).value_counts() # La mayoría es estacionaria
+
+# %%
+dificiles
```
@danpereda, Estos que  tienen indice mayor a cero no entendí que eran...

danpereda replied:
Mayor a cero significa que hay que diferenciar la serie para transformarla en estacionaria, hay una que tiene orden 3, significa que tienes que hacer .diff().dif().dif(), básicamente eso representa una serie de tiempo super loca porque cambia mucho en el tiempo.

datacubeR commented on notebooks_py/EDA_Target.py:
```diff
@@ -0,0 +1,160 @@
+
+reducer = umap.UMAP(n_components = 3)
+# El problema son los NaNs
+embedding = reducer.fit_transform(returns)
```
y este fallo!

danpereda replied:
Sí, aún no termino el EDA, pero me tuve que poner a hacer otras cosas jaja. 

Me queda eso y los test de hipótesis pendientes.

datacubeR reviewed (commented):
...

danpereda committed:
Update EDA_Target.ipynb

danpereda committed:
teoría

mirar la forma de los datos para ver qué tipo de clustering es mejor, igual son medios ellipsoidales, en una de esa es mejor usar mixturas gaussianas.

agregar plots de autocorrelación la idea es ver si tienen poder predictivo para agregar como feature.
